pennylane/ops/qubit/observables.py

```diff
@@ -435,13 +435,16 @@ class BasisStateProjector(Projector, Operation):
     # The call signature should be the same as Projector.__new__ for the positional
     # arguments, but with free key word arguments.
     def __init__(self, state, wires, id=None):
-        wires = Wires(wires)
-        state = list(qml.math.toarray(state).astype(int))
-
-        if not set(state).issubset({0, 1}):
-            raise ValueError(f"Basis state must only consist of 0s and 1s; got {state}")
+        wires = qml.wires.Wires(wires)
 
-        super().__init__(state, wires=wires, id=id)
+        if not qml.math.is_abstract(state):
+            # for cases like [jax.numpy.array(0), jax.numpy.array(1)]
+            state_array = qml.math.asarray(state)
+            state_array = state_array.astype(int)
+            if not all(x in [0, 1] for x in state_array):
+                raise ValueError(f"Basis state must only consist of 0s and 1s; got {state_array}")
+        else:
+            state_array = qml.math.asarray(state)
```

Comment:
```suggestion
            state_array = qml.math.asarray(state)
    super().__init__(state, wires=wires, id=id)
```

looks like the `super` acidentally got removed.